src/editor/ui_menus/main_menu_map_options.cc

```diff
@@ -36,6 +36,146 @@ inline EditorInteractive& MainMenuMapOptions::eia() {
 	return dynamic_cast<EditorInteractive&>(*get_parent());
 }
 
+constexpr unsigned kSuggestedTeamsUnitSize = 24;
+
+SuggestedTeamsEntry::SuggestedTeamsEntry(MainMenuMapOptions* mmmo,
+                                         UI::Panel* parent,
+                                         const Widelands::Map& map,
+                                         unsigned w,
+                                         Widelands::SuggestedTeamLineup t)
+   : UI::Panel(parent, 0, 0, w, kSuggestedTeamsUnitSize, _("Click player to remove")),
+     map_(map),
+     team_(t),
+     delete_(this,
+             "delete",
+             0,
+             0,
+             kSuggestedTeamsUnitSize,
+             kSuggestedTeamsUnitSize,
+             UI::ButtonStyle::kWuiSecondary,
+             _("Delete"),
+             _("Delete this suggested team lineup")) {
+	for (size_t index = 0; index < team_.size(); ++index) {
+		dropdowns_.push_back(create_dropdown(index));
+	}
+
+	delete_.sigclicked.connect([this, mmmo]() { mmmo->delete_suggested_team(this); });
+
+	update();
+}
+
+UI::Dropdown<Widelands::PlayerNumber>* SuggestedTeamsEntry::create_dropdown(size_t index) {
+	UI::Dropdown<Widelands::PlayerNumber>* dd = new UI::Dropdown<Widelands::PlayerNumber>(
+	   this, std::to_string(index), 0, index * kSuggestedTeamsUnitSize, kSuggestedTeamsUnitSize, 8,
+	   0, _("+"), UI::DropdownType::kPictorialMenu, UI::PanelStyle::kWui,
+	   UI::ButtonStyle::kWuiSecondary);
+	for (size_t i = 0; i < map_.get_nrplayers(); ++i) {
+		dd->add(map_.get_scenario_player_name(i + 1), i,
+		        playercolor_image(i, "images/players/player_position_menu.png"));
+	}
+	dd->set_tooltip(_("Add a player to this team"));
+	dd->selected.connect([this, dd]() {
+		const Widelands::PlayerNumber player = dd->get_selected();
+		// add this player to this team and remove him from all other teams in this lineup
+		for (Widelands::SuggestedTeam& t : team_) {
+			for (auto it = t.begin(); it != t.end(); ++it) {
+				if (*it == player) {
+					t.erase(it);
+					break;
+				}
+			}
+		}
+		// determine our index (it may have changed since the creation!)
+		int dd_index = -1;
+		for (size_t i = 0; i < dropdowns_.size(); ++i) {
+			if (dropdowns_[i] == dd) {
+				dd_index = i;
+				break;
+			}
+		}
+		assert(dd_index >= 0);
+		if (static_cast<unsigned>(dd_index) >= team_.size()) {
+			assert(static_cast<unsigned>(dd_index) == team_.size());
+			team_.push_back(Widelands::SuggestedTeam());
+		}
+		team_[dd_index].push_back(player);
+		update();
+	});
+	return dd;
+}
+
+void SuggestedTeamsEntry::layout() {
+	const uint16_t h = kSuggestedTeamsUnitSize * (team_.size() + 1);
+	set_desired_size(get_w(), h);
+	delete_.set_pos(Vector2i(get_w() / 2, h - kSuggestedTeamsUnitSize));
```

Comment:
Please put the delete button in a `UI::Box` - we want to move everything to Box layout, so we should not introduce new UI elements that don't use Box layout.

Same for the dropdowns.

Reply:
Dropdowns are dynamically deleted when they're not needed any more, and UI::Box can't cope with that can it?

Reply:
no, I don't think so, see https://github.com/widelands/widelands/pull/3847#issuecomment-660266974
Would of course be very appreciated, but might be rather complicated to implement...

Reply:
Let's keep things as they are in this branch then. I have opened an issue for fixing this problem in the future: https://github.com/widelands/widelands/issues/4062